File: Rust/Armstrong_Numbers.rs
pub fn is_armstrong_number(num: u32) -> bool 
{
    let mut sum = 0;
    for n in num.to_string().chars()
    {
        let mut x = n.to_digit(10).unwrap();
        x = x.pow(num.to_string().len() as u32);
        sum += x;
    }
    sum == num
}

File: C/armstrong_numbers.c
#include "armstrong_numbers.h"
#include <math.h>
bool is_armstrong_number(int candidate)
{
    int sum = 0;
    int digit_count = 0;
    int temp = candidate;
    
    while (temp != 0)
    {
        temp /= 10;
        digit_count++;
    }

    temp = candidate;
    while (temp != 0)
    {
        int digit = temp % 10;
        sum += pow(digit, digit_count);
        temp /= 10;
    }
    return sum == candidate;
}